### PR TITLE
[KOGITO-5300] Updating documentation of addons modules

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

In this PR, we add `README.md` files in every addon in `addons/commons` module. Also, the main README has been updated to reflect these changes.

In general, each doc has a short description of the addon, a link to the official doc, and examples where a user can see it in action.

If you're the original author and wish to add more information, feel free to add suggestions or send a PR directly to my branch so that we can merge everything once.

We are missing more information (and sometimes examples) for `human-task-prediction`, `mail`, and `filesystem` and `JDBC` persistence. So I've opened subtasks for this JIRA and assigned the SMEs. They are not a blocker for this one, though. See: https://issues.redhat.com/browse/KOGITO-5300

### CODE REVIEW

**Comment by ricardozanini** [comment]
@hmanwani-rh, can you please take a first look at the general structure of the documents and writing style? Also, please feel free to modify the files if you see some broken links or identify other sections in our docs that can be linked in theREADMEs.

**Comment by ricardozanini** [comment]
Adding @briandooley to the loop :)

File: addons/README.md [hunk 1]
```diff
@@ -76,9 +97,8 @@ knowledge of the Kogito engine internals. Then follow these steps:
 2. Do not use any dependencies from Quarkus or Spring Boot in `common` module. Your add-on must only have code to support
    your capability. New dependencies must be added to the [kogito-build](../kogito-build/kogito-build-parent) BOM.
 3. Create the same capability under the `addons` module in the runtime module you wish to add support (
-   either [`quarkus`](../quarkus/addons) or [`springboot`](../springboot/addons)). Make sure that your capability
-   imports either `quarkus-bom` or `springboot-dependencies` BOMs. You can choose to give support to only one of them,
-   just make it clear why and discuss this decision with the community.
+   either [`quarkus`](../quarkus/addons) or [`springboot`](../springboot/addons)). You can choose to give support to
```
Comment (briandooley) [inline addons/README.md:100]:
```suggestion
   [`quarkus`](../quarkus/addons) or [`springboot`](../springboot/addons)). You can give support to
```

File: addons/README.md [hunk 2]
```diff
@@ -76,9 +97,8 @@ knowledge of the Kogito engine internals. Then follow these steps:
...
+   only one of them, just make it clear why and discuss this decision with the community.
```
Comment (briandooley) [inline addons/README.md:101]:
```suggestion
   one add-on only. Make the reason for your decision clear and discuss this decision with the community.
```

File: addons/README.md [hunk 3]
```diff
@@ -87,8 +107,8 @@ knowledge of the Kogito engine internals. Then follow these steps:
 ### Add a new Capability Implementation to Kogito
 
 Sometimes, a capability requires a specific implementation. For example, `persistence`. It can be implemented by many
-persistence technologies that can their specific details. Each implementation can be handled differently based on the
-runtimes supported by Kogito:
+persistence technologies that can have their specific details. Each implementation can be handled differently based on
```
Comment (briandooley) [inline addons/README.md:110]:
```suggestion
persistence technologies that can have their specific details. Each implementation is handled differently based on
```

File: addons/common/cloudevents/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Kogito CloudEvents Addon
+
+Use this addon to make the events produced and consumed by a Kogito project accept
```
Comment (briandooley) [inline addons/common/cloudevents/README.md:3]: Decide whether to use "addon" or "add-on", throughout the entire document, for consistency.

File: addons/common/cloudevents/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+the [CloudEvents](https://cloudevents.io) format. There are many use cases where you will need to use this addon. Please
```
Comment (briandooley) [inline addons/common/cloudevents/README.md:4]:
```suggestion
the [CloudEvents](https://cloudevents.io) format. There are many use cases where you can use this addon.
```

File: addons/common/cloudevents/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+check the following sections in our documentation to find out more:
```
Comment (briandooley) [inline addons/common/cloudevents/README.md:5]:
```suggestion
See the following sections in our documentation to find out more:
```

File: addons/common/events/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# Kogito Events Addon
+
+Adds the capability to handle with events use cases on Kogito projects. Please find out more in the following sections
```
Comment (briandooley) [inline addons/common/events/README.md:3]:
```suggestion
The Kogito Events add-on adds the capability to handle use cases with events on Kogito projects. See the following documentation sections to find out more.
```

File: addons/common/events/README.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+of our official documentation:
```
Comment (briandooley) [inline addons/common/events/README.md:4]:
```suggestion
```

File: addons/common/events/README.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+- [Kogito runtime events](https://docs.jboss.org/kogito/release/latest/html_single/#con-kogito-runtime-events_kogito-configuring)
+- [Travel agency tutorial for Kogito services on OpenShift](https://docs.jboss.org/kogito/release/latest/html_single/#con-kogito-travel-agency_kogito-deploying-on-openshift)
+
+## Examples
+
+Please check the following examples to explore more about this capability.
```
Comment (briandooley) [inline addons/common/events/README.md:11]:
```suggestion
See the following examples to explore more about this capability.
```

File: addons/common/explainability/README.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# Kogito Explainability Addon
+
+This addon enables the integration with Explainability and Trusty Services. Please check out
```
Comment (briandooley) [inline addons/common/explainability/README.md:3]:
```suggestion
The Kogito Explainability Addon enables the integration with Explainability and Trusty Services. See 
```

File: addons/common/explainability/README.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+our [official documentation](https://docs.jboss.org/kogito/release/latest/html_single/#con-trusty-service_kogito-configuring)
+to find out more about it.
```
Comment (briandooley) [inline addons/common/explainability/README.md:5]:
```suggestion
to find out more.
```

File: addons/common/explainability/README.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+
+## Examples
+
+Please check the following example to explore more about this capability:
```
Comment (briandooley) [inline addons/common/explainability/README.md:9]:
```suggestion
See the following example to explore more about this capability:
```

File: addons/common/jobs/README.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
+# Kogito Jobs Management Add-on
+
+This add-on allows the integration and configuration of a Kogito project with the Jobs Service supporting service.
```
Comment (briandooley) [inline addons/common/jobs/README.md:3]:
```suggestion
Kogito Jobs Management Add-on allows the integration and configuration of a Kogito project with the Jobs Service supporting service.
```

File: addons/common/jobs/README.md [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+Please check
```
Comment (briandooley) [inline addons/common/jobs/README.md:4]:
```suggestion
```

File: addons/common/jobs/README.md [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+the [official documentation](https://docs.jboss.org/kogito/release/latest/html_single/#con-jobs-service_kogito-configuring)
```
Comment (briandooley) [inline addons/common/jobs/README.md:5]:
```suggestion
See the [official documentation](https://docs.jboss.org/kogito/release/latest/html_single/#con-jobs-service_kogito-configuring)
```

File: addons/common/jobs/README.md [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+to find out more.
+
+## Examples
+
+Please check the following examples to explore more about this capability:
```
Comment (briandooley) [inline addons/common/jobs/README.md:10]:
```suggestion
See the following examples to explore more about this capability:
```

File: addons/common/mail/README.md [hunk 17]
```diff
@@ -0,0 +1,13 @@
+# Kogito Mail Addon
+
+This addon enables the capability of sending emails in a Kogito Process project. An example of usage is the Kogito Task
```
Comment (briandooley) [inline addons/common/mail/README.md:3]:
```suggestion
The Kogito Mail Addon adds the capability of sending emails in a Kogito Process project. An example of this is the Kogito Task
```

File: addons/common/mail/README.md [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+Deadline use case. <!-- TODO: reference needed -->
+
+<!-- TODO: this addon is lacking documentation in the official docs. Please see: https://issues.redhat.com/browse/KOGITO-5562 -->
+
+## Examples
+
+Please check the following examples to explore more about this capability:
```
Comment (briandooley) [inline addons/common/mail/README.md:10]:
```suggestion
See the following examples to explore more about this addon:
```

File: addons/common/monitoring/README.md [hunk 19]
```diff
@@ -0,0 +1,16 @@
+# Kogito Monitoring Addons
+
+These are the addons for adding monitoring capabilities to Kogito projects. Please check
```
Comment (briandooley) [inline addons/common/monitoring/README.md:3]:
```suggestion
The Kogito Monitoring Addons add monitoring capabilities to Kogito projects. See 
```

File: addons/common/persistence/README.md [hunk 20]
```diff
@@ -0,0 +1,19 @@
+# Kogito Persistence Addons
+
+These addons add the persistence capability to Kogito projects. Please check
```
Comment (briandooley) [inline addons/common/persistence/README.md:3]:
```suggestion
The Kogito Persistence Addons add persistence capability to Kogito projects.See 
```

File: addons/common/process-management/README.md [hunk 21]
```diff
@@ -0,0 +1,21 @@
+# Kogito Process Management Addon
+
+This addon adds the capability of process management to Kogito project and is also part of the integration with the
```
Comment (briandooley) [inline addons/common/process-management/README.md:3]:
```suggestion
The Kogito Process Management Addons add process management capability to Kogito projects and are also part of the integration with the
```

File: addons/common/process-management/README.md [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+[Kogito Management Console](https://github.com/kiegroup/kogito-apps/tree/master/management-console). Please check
```
Comment (briandooley) [inline addons/common/process-management/README.md:4]:
```suggestion
[Kogito Management Console](https://github.com/kiegroup/kogito-apps/tree/master/management-console). See
```

File: addons/common/process-management/README.md [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+the [official documentation](https://docs.jboss.org/kogito/release/latest/html_single/#con-bpmn-process-management-addon_kogito-developing-process-services)
+to find out more!
```
Comment (briandooley) [inline addons/common/process-management/README.md:6]:
```suggestion
to find out more.
```

File: addons/common/process-management/README.md [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+
+## Examples
+
+Please check the following examples to explore more about this capability:
```
Comment (briandooley) [inline addons/common/process-management/README.md:10]:
```suggestion
See the following examples to explore more about this capability:
```

File: addons/common/process-svg/README.md [hunk 25]
```diff
@@ -0,0 +1,16 @@
+# Kogito Process SVG Addon
+
+This addon adds the capability to enable SVG visualization to process diagrams. Please check
```
Comment (briandooley) [inline addons/common/process-svg/README.md:3]:
```suggestion
The Kogito Process SVG Addon adds the capability to enable SVG visualization to process diagrams. See
```

File: addons/common/process-svg/README.md [hunk 26]
```diff
@@ -0,0 +1,16 @@
...
+the [official documentation](https://docs.jboss.org/kogito/release/latest/html_single/#con-bpmn-process-svg-addon_kogito-developing-process-services)
+to find out more.
+
+## Examples
+
+Please check the following examples to explore more about this capability:
```
Comment (briandooley) [inline addons/common/process-svg/README.md:9]:
```suggestion
See the following examples to explore more about this capability:
```

File: addons/common/rest-exception-handler/README.md [hunk 27]
```diff
@@ -0,0 +1,3 @@
+# Kogito REST Exception Handler Addon
+
+This addon is supposed to be used internally by other addons. It's not meant to be used directly by end-user's projects.
```
Comment (briandooley) [inline addons/common/rest-exception-handler/README.md:3]:
```suggestion
The Kogito REST Exception Handler Addon is used internally by other addons. It is not to be used directly by end-user's projects.
```

File: addons/common/task-management/README.md [hunk 28]
```diff
@@ -0,0 +1,13 @@
+# Kogito Task Management Addon
+
+Enables the integration with the Kogito supporting
```
Comment (briandooley) [inline addons/common/task-management/README.md:3]:
```suggestion
The Kogito Task Management Addon enables integration with the Kogito supporting 
```

File: addons/common/task-management/README.md [hunk 29]
```diff
@@ -0,0 +1,13 @@
...
+service [Task Console](https://github.com/kiegroup/kogito-apps/tree/master/task-console). Please check
```
Comment (briandooley) [inline addons/common/task-management/README.md:4]:
```suggestion
service [Task Console](https://github.com/kiegroup/kogito-apps/tree/master/task-console). See 
```

File: addons/common/task-management/README.md [hunk 30]
```diff
@@ -0,0 +1,13 @@
...
+the [official documentation](https://docs.jboss.org/kogito/release/latest/html_single/#con-task-console_kogito-developing-process-services)
+to find out more!
```
Comment (briandooley) [inline addons/common/task-management/README.md:6]:
```suggestion
to find out more.
```

File: addons/common/task-management/README.md [hunk 31]
```diff
@@ -0,0 +1,13 @@
...
+
+## Examples
+
+Please check the following examples to explore more about this capability:
```
Comment (briandooley) [inline addons/common/task-management/README.md:10]:
```suggestion
See the following examples to explore more about this capability:
```

File: addons/common/tracing/README.md [hunk 32]
```diff
@@ -0,0 +1,12 @@
+# Kogito Tracing Addon
+
+Adds the capability of tracing to Kogito projects. At this time, we only support DMN services tracing. To find out more
```
Comment (briandooley) [inline addons/common/tracing/README.md:3]:
```suggestion
The Kogito Tracing addon adds tracing capability to Kogito projects. Currently, we support DMN services tracing only. To find out more
```

File: addons/common/tracing/README.md [hunk 33]
```diff
@@ -0,0 +1,12 @@
...
+about this capability, please check
```
Comment (briandooley) [inline addons/common/tracing/README.md:4]:
```suggestion
about this capability, see 
```

File: addons/common/tracing/README.md [hunk 34]
```diff
@@ -0,0 +1,12 @@
...
+our [official documentation](https://docs.jboss.org/kogito/release/latest/html_single/#con-trusty-service_kogito-configuring).
+
+## Examples
+
+Please check the following examples to explore more about this capability:
```
Comment (briandooley) [inline addons/common/tracing/README.md:9]:
```suggestion
See the following examples to explore more about this capability:
```

**Review by briandooley — CHANGES_REQUESTED** [review]

File: addons/README.md [hunk 35]
```diff
@@ -1,7 +1,28 @@
 # Kogito Add-Ons
 
-In this package you will find the base Addons modules that extend the Kogito Core capabilities. They are responsible to
-give persistence, monitoring, messaging, and many other features on top of a plain Kogito service.
+In this package you will find the Add-Ons modules that extend the Kogito Core capabilities. They add 
+persistence, monitoring, messaging, and many other features to the Kogito service.
+
+## Add-Ons Catalog
+
+The following table lists all the add-onss, independent of runtime implementation, supported by the Kogito Community:
```
Comment (MarianMacik) [inline addons/README.md:8]:
```suggestion
The following table lists all the add-ons, independent of runtime implementation, supported by the Kogito Community:
```

File: addons/README.md [hunk 36]
```diff
@@ -59,8 +80,8 @@ In general, the add-on name is composed like `kogito-addons-{runtime}-{capabilit
   the capability. It's more a way to differentiate the implementations.
 
 The core add-ons don't have a `runtime` prefix either because they are the bases for the implementation by a runtime, or
-it is a generic implementation that can work either on Quarkus or Spring Boot. Please, check the documentation of the
-add-on to find out.
+it is a generic implementation that can work either on [Quarkus](../quarkus/addons) or [Spring Boot](../quarkus/addons).
```
Comment (MarianMacik) [inline addons/README.md:83]: Probably `they are a generic implementation` just to keep it uniform.

File: addons/common/cloudevents/README.md [hunk 37]
```diff
@@ -0,0 +1,26 @@
+# Kogito CloudEvents Add-on
+
+Use this addon to make the events produced and consumed by a Kogito project accept
```
Comment (MarianMacik) [inline addons/common/cloudevents/README.md:3]:
```suggestion
Use this add-on to make the events produced and consumed by a Kogito project accept
```

File: addons/common/cloudevents/README.md [hunk 38]
```diff
@@ -0,0 +1,26 @@
...
+the [CloudEvents](https://cloudevents.io) format. There are many use cases where you can use this addon.
```
Comment (MarianMacik) [inline addons/common/cloudevents/README.md:4]:
```suggestion
the [CloudEvents](https://cloudevents.io) format. There are many use cases where you can use this add-on.
```

File: addons/common/monitoring/README.md [hunk 39]
```diff
@@ -0,0 +1,16 @@
+# Kogito Monitoring Add-ons
+
+The Kogito Monitoring Add-ons add monitoring capabilities to Kogito projects. See
```
Comment (MarianMacik) [inline addons/common/monitoring/README.md:3]:
```suggestion
The Kogito Monitoring Add-ons adds monitoring capabilities to Kogito projects. See 
```

File: addons/common/persistence/README.md [hunk 40]
```diff
@@ -0,0 +1,19 @@
+# Kogito Persistence Add-ons
+
+The Kogito Persistence Add-ons add persistence capability to Kogito projects.See
```
Comment (MarianMacik) [inline addons/common/persistence/README.md:3]:
```suggestion
The Kogito Persistence Add-ons adds persistence capability to Kogito projects.See 
```

File: addons/common/process-management/README.md [hunk 41]
```diff
@@ -0,0 +1,20 @@
+# Kogito Process Management Add-on
+
+The Kogito Process Management Add-ons add process management capability to Kogito projects and are also part of the integration with the
```
Comment (MarianMacik) [inline addons/common/process-management/README.md:3]:
```suggestion
The Kogito Process Management Add-ons adds process management capability to Kogito projects and are also part of the integration with the
```

**Review by MarianMacik — APPROVED** [review]
Looks good, just some typos.

**Review by briandooley — APPROVED** [review]
LGTM

**Pull request merged**